libcxx/include/__ranges/helpers.h

```diff
@@ -0,0 +1,39 @@
+// -*- C++ -*-
```

Comment:
> Otherwise it could just live in `join_view.h`.

Hmm, according to the paper, _`as-lvalue`_ should be shared between `join_view` and `join_with_view`, and thus it deseves a separated internal header.

Reply:
Sounds reasonable, what do you think about `__utility/as_lvalue.h`?

Reply:
I've moved *`as-lvalue`* to `__utility/as_lvalue.h`, but I've left it in `ranges` namespace since it is used only by `join` and `join_with` views.